compiler/crates/relay-typegen/src/visit.rs

```diff
@@ -2461,6 +2467,82 @@ fn transform_non_nullable_input_type(
     }
 }
 
+fn build_input_object(
+    typegen_context: &TypegenContext<'_>,
+    input_object: &InputObject,
+    input_object_types: &mut InputObjectTypes,
+    encountered_enums: &mut EncounteredEnums,
+    custom_scalars: &mut CustomScalarsImports,
+) -> ExactObject {
+    ExactObject::new(
+        input_object
+            .fields
+            .iter()
+            .map(|field| {
+                Prop::KeyValuePair(KeyValuePairProp {
+                    key: field.name.item.0,
+                    read_only: false,
+                    optional: !field.type_.is_non_null()
+                        || typegen_context
+                            .project_config
+                            .typegen_config
+                            .optional_input_fields
+                            .contains(&field.name.item.0)
+                        || field.default_value.is_some(),
+                    value: transform_input_type(
+                        typegen_context,
+                        &field.type_,
+                        input_object_types,
+                        encountered_enums,
+                        custom_scalars,
+                    ),
+                })
+            })
+            .collect(),
+    )
+}
+
+fn build_one_of_cases(
+    typegen_context: &TypegenContext<'_>,
+    input_object: &InputObject,
+    input_object_types: &mut InputObjectTypes,
+    encountered_enums: &mut EncounteredEnums,
+    custom_scalars: &mut CustomScalarsImports,
+) -> impl Iterator<Item = ExactObject> {
+    (0..input_object.fields.len())
+        .map(|present_field_idx| {
+            input_object
+                .fields
+                .iter()
+                .enumerate()
+                .map(|(idx, field)| {
+                    if idx == present_field_idx {
+                        Prop::KeyValuePair(KeyValuePairProp {
+                            key: field.name.item.0,
+                            read_only: false,
+                            optional: false,
+                            value: transform_non_nullable_input_type(
+                                typegen_context,
+                                &field.type_,
+                                input_object_types,
+                                encountered_enums,
+                                custom_scalars,
+                            ),
+                        })
+                    } else {
+                        Prop::KeyValuePair(KeyValuePairProp {
+                            key: field.name.item.0,
+                            read_only: false,
+                            optional: true,
+                            value: AST::Empty,
```

Comment:
Can you share a bit of context (here, and ideally in a comment) on why we choose to include these fields but type them as `empty`/`never` instead of simply omitting them from the union variant?

Reply:
If it's a considered choice with benefits, I'm fine with it. I just wasn't sure why we had made that choice.